kernel/src/hil/screen.rs

```diff
@@ -116,6 +135,22 @@ impl ScreenPixelFormat {
     }
 }
 
+/// Placement of a grid of rectangles on the display.
+pub struct Grid {
+    /// Placement of a vertical edge of a rectangle
+    /// relative to the left edge of the display.
+    /// Additional rectangles are tiled to the left and right
+    /// without overlapping, until they fill the entire display area.
+    /// Greater values are further to the right.
+    pub u16: x_offset,
+    /// Greater values are further downwards.
+    pub u16: y_offset,
```

Comment:
I understand `width` and `height`, but I don't quite get the meaning of the offsets.

Reply:
I can imagine screens where the dimension is not a whole multiple of the tile size, putting the *end* at a different offset. From there, it's not a big jump to assume an odd start, perhaps due to rotation. I have no examples though.

Reply:
I would assume that the end would have some ignored pixels. Most probably the memory space would be there (as usually tiles are related to memory units), but the pixels are missing. I would suggest having only `width` and `height`.

Reply:
What should be done for displays rotated 180°?

Reply:
> Would there ever be a screen where the tiling didn't start at 0,0?

Yeah, I've seen this done. It happens when you want the screen to be a window into a larger framebuffer. E.g., you could do something where you load a framebuffer with a 1000x1000 image, the screen is 200x200, and you want to tile a 100x100 patch of the framebuffer 4 times into the screen.

Reply:
> What should be done for displays rotated 180°?

I think you want to operate in framebuffer space. Suppose I have a device that can detect orientation and wants to adjust the screen rotation accordingly (e.g., a modern phone). When the device rotates 180 degrees, I continue to draw up as up (the top left of the image is at 0, 0) but then I physically rotate it 180 degrees on the screen. You would also need to do geometric inversion for other inputs (e.g., a capacitive display, accelerometers), but keying this on a global rotation bit is easier than trying to dynamically switch draw calls.